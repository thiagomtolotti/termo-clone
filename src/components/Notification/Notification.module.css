.container {
  width: 100%;
  height: 50px;

  display: flex;
  align-items: center;
  justify-content: center;

  position: relative;
}

.notification {
  font-size: 1.3em;
  background-color: var(--blue);

  padding: 12px 36px;

  border-radius: 12px;

  position: absolute;

  animation: notification 200ms ease-in-out;
}

@keyframes notification {
  0% {
    transform: scale(0.2);
  }
}
