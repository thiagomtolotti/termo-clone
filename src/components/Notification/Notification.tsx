import styles from "./Notification.module.css";

export const Notification = () => {
  return <div className={styles.notification}>Notificação de teste</div>;
};
