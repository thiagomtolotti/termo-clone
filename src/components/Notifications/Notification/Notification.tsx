import { useEffect, useState } from "react";
import styles from "../Notifications.module.css";

interface NotificationProps {
  children: string;
}

export const Notification = ({ children }: NotificationProps) => {
  const [render, setRenderAgain] = useState(false);

  useEffect(() => {
    setRenderAgain(false);

    if (!children) return;

    setTimeout(() => setRenderAgain(true), 0);
  }, [children]);

  return render && <div className={styles.notification}>{children}</div>;
};
