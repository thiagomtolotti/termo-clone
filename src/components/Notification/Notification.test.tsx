import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Notification } from "./Notification";
import {
  ApplicationContext,
  IApplicationContext,
} from "../../context/ApplicationContext";
import React from "react";

jest.mock("../../context/ApplicationContext", () => ({
  __esModule: true,
  ApplicationContext: jest
    .requireActual("react")
    .createContext({} as IApplicationContext),
}));

let renderResult: HTMLElement;

const currentNotificationMock = {
  message: "Teste",
  timestamp: Date.now(),
};

beforeEach(() => {
  const { container } = render(
    <ApplicationContext.Provider
      value={
        {
          Notifications: {
            currentNotification: currentNotificationMock,
          },
        } as IApplicationContext
      }
    >
      <Notification />
    </ApplicationContext.Provider>
  );

  renderResult = container;
});

afterEach(() => {
  jest.clearAllMocks();
});

it("Should render correctly", () => {
  expect(renderResult).toMatchSnapshot();
});

it("Should render with the notification provided by the context", () => {
  const notificationEl = screen.getByRole("notification");

  expect(notificationEl.innerHTML).toBe(currentNotificationMock.message);
});
