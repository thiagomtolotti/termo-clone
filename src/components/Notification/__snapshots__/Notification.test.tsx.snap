// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div>
  <div
    class="w-full h-14 flex-shrink-0 flex items-center justify-center"
  >
    <div
      class="bg-blue py-3 px-9 rounded-xl text-xl notification "
      role="notification"
    >
      Teste
    </div>
  </div>
</div>
`;
